tf: standalone apm server aws_os not exist

regression from https://github.com/elastic/apm-server/pull/15535

```
│ Error: Invalid index
│ 
│   on ../infra/terraform/modules/standalone_apm_server/main.tf line 72, in data "aws_ami" "os":
│   72:     values = [local.instance_arch[var.aws_os]]
│     ├────────────────
│     │ local.instance_arch is object with 7 attributes
│     │ var.aws_os is "amzn2-ami-hvm-*-x86_64-ebs"
│ 
│ The given key does not identify an element in this collection value.
╵
╷
│ Error: Invalid index
│ 
│   on ../infra/terraform/modules/standalone_apm_server/main.tf line 85, in data "aws_ami" "os":
│   85:   owners = [local.image_owners[var.aws_os]]
│     ├────────────────
│     │ local.image_owners is object with 7 attributes
│     │ var.aws_os is "amzn2-ami-hvm-*-x86_64-ebs"
│ 
│ The given key does not identify an element in this collection value.
╵
```

Benchmarking on standalone apm server is still broken on main
```
╷
│ Error: creating EC2 Instance: InvalidParameterValue: The architecture 'x86_64' of the specified instance type does not match the architecture 'arm64' of the specified AMI. Specify an instance type and an AMI that have matching architectures, and try again. You can use 'describe-instance-types' or 'describe-images' to discover the architecture of the instance type or AMI.
│ 	status code: 400, request id: 68dbc0d4-fa89-46f6-96e7-2a17d2df0179
│ 
│   with module.standalone_apm_server[0].aws_instance.apm,
│   on ../infra/terraform/modules/standalone_apm_server/main.tf line 142, in resource "aws_instance" "apm":
│  142: resource "aws_instance" "apm" {
│ 
╵
make: *** [Makefile:92: apply] Error 1
```

--- a/testing/benchmark/main.tf
+++ b/testing/benchmark/main.tf
@@ -153,7 +153,7 @@ module "standalone_apm_server" {
   source = "../infra/terraform/modules/standalone_apm_server"
 
   vpc_id              = module.vpc.vpc_id
-  aws_os              = "al2023-ami-2023"
+  aws_os              = "al2023-ami-2023.*-x86_64"
   apm_instance_type   = var.standalone_apm_server_instance_size
   apm_volume_type     = var.standalone_apm_server_volume_type
   apm_volume_size     = var.apm_server_tail_sampling ? coalesce(var.standalone_apm_server_volume_size, 60) : var.standalone_apm_server_volume_size

--- a/testing/infra/terraform/modules/standalone_apm_server/main.tf
+++ b/testing/infra/terraform/modules/standalone_apm_server/main.tf
@@ -5,6 +5,7 @@ locals {
     "ubuntu-noble-24.04-arm64-server" = "099720109477" # canonical
     "debian-12-arm64"                 = "136693071363" # debian
     "al2023-ami-2023"                 = "137112412989" # amazon
+    "al2023-ami-2023.*-x86_64"        = "137112412989" # amazon
     "RHEL-8"                          = "309956199498" # Red Hat
     "RHEL-9"                          = "309956199498" # Red Hat
   }
@@ -14,6 +15,7 @@ locals {
     "ubuntu-noble-24.04-arm64-server" = "t4g.nano"
     "debian-12-arm64"                 = "t4g.nano"
     "al2023-ami-2023"                 = "t4g.nano"
+    "al2023-ami-2023.*-x86_64"        = "t3a.micro"
     "RHEL-8"                          = "t4g.micro" # RHEL doesn't support nano instances
     "RHEL-9"                          = "t4g.micro" # RHEL doesn't support nano instances
   }
@@ -23,6 +25,7 @@ locals {
     "ubuntu-noble-24.04-arm64-server" = "arm64"
     "debian-12-arm64"                 = "arm64"
     "al2023-ami-2023"                 = "arm64"
+    "al2023-ami-2023.*-x86_64"        = "x86_64"
     "RHEL-8"                          = "arm64"
     "RHEL-9"                          = "arm64"
   }
@@ -32,6 +35,7 @@ locals {
     "ubuntu-noble-24.04-arm64-server" = "curl ${data.external.latest_elastic_agent.result.deb_arm} -o elastic-agent.deb && sudo dpkg -i elastic-agent.deb"
     "debian-12-arm64"                 = "curl ${data.external.latest_elastic_agent.result.deb_arm} -o elastic-agent.deb && sudo dpkg -i elastic-agent.deb"
     "al2023-ami-2023"                 = "curl ${data.external.latest_elastic_agent.result.rpm_arm} -o elastic-agent.rpm && sudo yum -y install elastic-agent.rpm"
+    "al2023-ami-2023.*-x86_64"        = "curl ${data.external.latest_elastic_agent.result.rpm_arm} -o elastic-agent.rpm && sudo yum -y install elastic-agent.rpm"
     "RHEL-8"                          = "curl ${data.external.latest_elastic_agent.result.rpm_arm} -o elastic-agent.rpm && sudo yum -y install elastic-agent.rpm"
     "RHEL-9"                          = "curl ${data.external.latest_elastic_agent.result.rpm_arm} -o elastic-agent.rpm && sudo yum -y install elastic-agent.rpm"
   }
@@ -41,6 +45,7 @@ locals {
     "ubuntu-noble-24.04-arm64-server" = "curl ${data.external.latest_apm_server.result.deb_arm} -o apm-server.deb && sudo dpkg -i apm-server.deb"
     "debian-12-arm64"                 = "curl ${data.external.latest_apm_server.result.deb_arm} -o apm-server.deb && sudo dpkg -i apm-server.deb"
     "al2023-ami-2023"                 = "curl ${data.external.latest_apm_server.result.rpm_arm} -o apm-server.rpm && sudo yum -y install apm-server.rpm"
+    "al2023-ami-2023.*-x86_64"        = "curl ${data.external.latest_apm_server.result.rpm_arm} -o apm-server.rpm && sudo yum -y install apm-server.rpm"
     "RHEL-8"                          = "curl ${data.external.latest_apm_server.result.rpm_arm} -o apm-server.rpm && sudo yum -y install apm-server.rpm"
     "RHEL-9"                          = "curl ${data.external.latest_apm_server.result.rpm_arm} -o apm-server.rpm && sudo yum -y install apm-server.rpm"
   }
@@ -50,6 +55,7 @@ locals {
     "ubuntu-noble-24.04-arm64-server" = "ubuntu"
     "debian-12-arm64"                 = "admin"
     "al2023-ami-2023"                 = "ec2-user"
+    "al2023-ami-2023.*-x86_64"        = "ec2-user"
     "RHEL-8"                          = "ec2-user"
     "RHEL-9"                          = "ec2-user"
   }

